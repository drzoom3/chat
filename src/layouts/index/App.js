//libs
import React from 'react'
import PropTypes from 'prop-types'

import {
    Route,
    Switch
} from 'react-router-dom'

import {
    connect
} from 'react-redux'

import '../../elements/layout/layout.styl'

import NotFoundPage from '../../pages/NotFound'
import HomePage from '../../pages/Home'
import ChatPage from '../../pages/Chat'

import Header from '../../components/Header'
import Footer from '../../components/Footer'

class App extends React.Component {
    constructor(props) {
        super(props);

        this.state = {}
    }

    render() {        
        return <div className='body'>
            <Header/>
            <Switch>
                <Route exact path='/' component={ HomePage } />
                <Route exact path='/:chatID' component={ ChatPage } />
                <Route component={ NotFoundPage } />    
            </Switch>
            <Footer/>
        </div>
    }
}

if (process.env.NODE_ENV == 'development') {
    App.propTypes = {
    };
}

export default connect(
    state => ({
    }),
    dispatch => ({
    })
)(App);