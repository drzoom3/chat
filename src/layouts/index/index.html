<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="theme-color" content="#fc8f02">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <meta name="msapplication-starturl" content="/">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no, shrink-to-fit=no">
    <meta name="keywords" content="" />
    <meta name="description" content="" />
    <meta property="og:url" content="" />
    <meta property="og:type" content="website" />
    <meta property="og:title" content="" />
    <meta property="og:description" content="" />
    <meta property="og:image" content="" />
    <title></title>
    <link href="/favicon.ico" rel="shortcut icon" type="image/x-icon" />
    <script>var $INJECT_CONFIG;</script>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>