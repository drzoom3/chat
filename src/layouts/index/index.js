//libs
import React from 'react'
import { render } from 'react-dom'
import cookie from 'react-cookie'
import {
    BrowserRouter,
    Route
} from 'react-router-dom'
import {
    Provider
} from 'react-redux'

import {
    store
} from '../../utils/reducers'//store

import App from './App'

render(
    <Provider store={store} >
        <BrowserRouter>
            <Route path='/' component={App} />
        </BrowserRouter>
    </Provider>,
    document.getElementById('root')
)
