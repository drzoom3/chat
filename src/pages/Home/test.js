import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import { Home } from './index'
import chats from '../../../_mocks/state-chats'

describe('Home', () => {
    it('Render', () => {
        const renderer = new ShallowRenderer();
        const store = {
            chats
        }
        const getChatsListSuccess = jest.fn();
        getChatsListSuccess.mockReturnValue();
        
        const addChatSuccess = jest.fn();
        addChatSuccess.mockReturnValue();

        const history = {
            push: jest.fn()
        }
        
        const result = renderer.render(<Home
            { ...store }
            history={ history }
            getChatsListSuccess={ getChatsListSuccess }
            addChatSuccess={ addChatSuccess }
        />)

        expect(result).toMatchSnapshot();
    });
});