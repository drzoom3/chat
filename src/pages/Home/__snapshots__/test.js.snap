// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Render 1`] = `
<div
  className="page page--home"
>
  <div
    className="container"
  >
    <div>
      <div>
        <Link
          replace={false}
          to="/11"
        >
          11
        </Link>
      </div>
    </div>
    <button
      onClick={[Function]}
    >
      Создать комнату
    </button>
  </div>
</div>
`;
